File: src/main/java/net/frozenorb/potpvp/util/menu/buttons/BooleanButton.java
package net.frozenorb.potpvp.util.menu.buttons;

import net.frozenorb.potpvp.util.Callback;
import net.frozenorb.potpvp.util.menu.Button;
import lombok.AllArgsConstructor;
import org.bukkit.Material;
import org.bukkit.Sound;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.ClickType;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
public class BooleanButton extends Button {

    private boolean confirm;
    private Callback<Boolean> callback;

    public void clicked(Player player, int i, ClickType clickType) {

        if (this.confirm) {
            player.playSound(player.getLocation(), Sound.NOTE_PIANO, 20.0F, 0.1F);
        } else {
            player.playSound(player.getLocation(), Sound.DIG_GRAVEL, 20.0F, 0.1F);
        }

        player.closeInventory();

        this.callback.callback(this.confirm);
    }

    @Override
    public String getName(Player player) {
        return this.confirm ? "§aConfirm" : "§cCancel";
    }

    @Override
    public List<String> getDescription(Player player) {
        return new ArrayList();
    }

    @Override
    public byte getDamageValue(Player player) {
        return (byte)(this.confirm ? 5 : 14);
    }

    @Override
    public Material getMaterial(Player player) {
        return Material.WOOL;
    }}

File: src/main/kotlin/net/frozenorb/potpvp/kt/menu/buttons/BooleanButton.kt
package net.frozenorb.potpvp.kt.menu.buttons

import net.frozenorb.potpvp.kt.menu.Button
import org.bukkit.Material
import org.bukkit.entity.Player
import java.util.ArrayList
import org.bukkit.event.inventory.ClickType
import org.bukkit.inventory.InventoryView

class BooleanButton(private val confirm: Boolean, private val callback: (Boolean) -> Unit) : Button() {

    override fun getName(player: Player): String {
        return if (confirm) "§aConfirm" else "§cCancel"
    }

    override fun getDescription(player: Player): List<String> {
        return ArrayList()
    }

    override fun getDamageValue(player: Player): Byte {
        return (if (this.confirm) 5 else 14).toByte()
    }

    override fun getMaterial(player: Player): Material {
        return Material.WOOL
    }

    override fun clicked(player: Player, i: Int, clickType: ClickType, view: InventoryView) {
        if (confirm) {
            playSuccess(player)
        } else {
            playFail(player)
        }

        player.closeInventory()

        callback.invoke(confirm)
    }

}